Mathlib/LinearAlgebra/BilinearMap.lean

```diff
@@ -429,6 +429,49 @@ open Function
 
 section restrictScalarsRange
 
+variable {R S M P M' P' : Type*}
+  [CommSemiring R] [CommSemiring S] [SMul S R]
+  [AddCommMonoid M] [Module R M] [AddCommMonoid P] [Module R P]
+  [Module S M] [Module S P]
+  [IsScalarTower S R M] [IsScalarTower S R P]
+  [AddCommMonoid M'] [Module S M'] [AddCommMonoid P'] [Module S P']
+
+variable (i : M' →ₗ[S] M) (k : P' →ₗ[S] P) (hk : Injective k)
+  (f : M →ₗ[R] P) (hf : ∀ m, f (i m) ∈ LinearMap.range k)
+
+include hf in
+theorem restrictScalars_in_range (m : M') : (f.restrictScalars S).comp i m ∈ Set.range k := by
+  use (hf m).choose
+  rw [(hf m).choose_spec, comp_apply, restrictScalars_apply]
+
```

Comment:
The proof here can be just `exact hf m`
```suggestion
```
so I think I'd just inline the below as indicated.